src/components/MultiInput.vue

```diff
@@ -0,0 +1,114 @@
+<template>
+	<CdsFlexbox
+		direction="column"
+		gap="3"
+	>
+		<CdsFlexbox
+			v-for="(item, index) in internalModel"
+			:key="item.id"
+			align="center"
+			gap="3"
+		>
+			<CdsTextInput
+				v-model="item.value"
+				floating-label
+				:label="inputLabel"
+			/>
+
+			<CdsClickable
+				v-if="internalModel.length > 1"
+				clickable
+			>
+				<CdsIcon
+					height="20"
+					width="20"
+					name="x-outline"
+					@click="removeInput(index)"
+				/>
+			</CdsClickable>
+		</CdsFlexbox>
+
+		<CdsFlatButton
+			:variant="variant"
+			@click="addInput"
+		>
+			<CdsFlexbox
+				align="center"
+				gap="2"
+			>
+				<CdsIcon
+					height="16"
+					width="16"
+					name="plus-outline"
+				/>
+
+				{{ buttonLabel }}
+			</CdsFlexbox>
+		</CdsFlatButton>
+	</CdsFlexbox>
+</template>
+
+<script setup>
+import { defineProps, ref, watch } from 'vue';
+import CdsTextInput from './TextInput.vue';
+import CdsIcon from './Icon.vue';
+import CdsFlexbox from './Flexbox.vue';
+import CdsFlatButton from './FlatButton.vue';
+import CdsClickable from './Clickable.vue';
+import generateKey from '../utils/methods/uuidv4';
+
+const model = defineModel('modelValue', {
+	type: Array,
+	default: () => [
+		{
+			id: generateKey(),
+			value: '',
+		}
+	],
+});
+
+defineProps({
+	inputLabel: {
+		type: String,
+		default: 'Nova opção',
+	},
+	buttonLabel: {
+		type: String,
+		default: 'Adicionar',
+	},
+	variant: {
+		type: String,
+		default: 'dark',
+	},
+});
+
+
+const internalModel = ref([]);
+
+watch(model, () => {
+	internalModel.value = model.value;
+}, { immediate: true });
+
+watch(internalModel, () => {
+	model.value = internalModel.value;
+});
+
+function addInput() {
+	internalModel.value.push({
```

Comment:
- Dar um "focus" no input novo
- Adicionar suporte ao enter
- Ter uma opção para remover label (só passar vazio ficar estranho pq a label continua 'floating', só q vazio)

Sugestões:
- Desabilitar o botão quando a última opção é vazia (para evitar que o usuário adicione diversas opções vazias desnecessariamente)
- Adicionar a possibilidade de o usuário poder confirmar antes de deletar da lista
- Como é um input, eu adicionaria a possibilidade de uma label 'por fora' (Ex.: Checklist, Lista de opções)
- Plus plus plus: Drag and drop para ordenar a lista

Reply:
Implementado:
- "Dar um "focus" no input novo";
- "Adicionar suporte ao enter;
- "Desabilitar o botão quando a última opção é vazia";
- "Como é um input, eu adicionaria a possibilidade de uma label 'por fora'"